Documentation/Basic-Usage.md

```diff
@@ -47,57 +61,88 @@ struct User {
-- [Introduction to Function Currying in Swift](http://robots.thoughtbot.com/introduction-to-function-currying-in-swift)
+> If you'd like to learn more about currying, we recommend the following articles:
+> 
+> 
+> - [Apple's documentation of curried functions](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Declarations.html#//apple_ref/doc/uid/TP40014097-CH34-XID_615)
+> - [Introduction to Function Currying in Swift](http://robots.thoughtbot.com/introduction-to-function-currying-in-swift)
 
 Now, we make `User` conform to `Decodable` and implement the required `decode`
-function. We will implement this function by using `map` (`<^>`) and `apply`
-(`<*>`) to conditionally pass the required parameters to the curried init
-function. The common pattern will look like:
+function. We will implement this function by using some [functional
+concepts](functional_concepts),
+specifically the `map` (`<^>`) and `apply` (`<*>`) operators, to conditionally
+pass the required parameters to the curried init function. The common pattern
+will look like this:
+
+[functional_concepts]: https://github.com/thoughtbot/Argo/blob/master/Documentation/Functional-Concepts.md
 
 ```swift
-return curry(Model.init) <^> paramOne <*> paramTwo <*> paramThree
+  static func decode(json: JSON) -> Decoded<DecodedType> {
+    return curry(Model.init) <^> paramOne <*> paramTwo <*> paramThree
+  }
 ```
 
-and so on. If any of those parameters are an error, the entire creation process
-will fail, and the function will return the first error. If all of the
-parameters are successful, the value will be unwrapped and passed to the
-`init` function.
+and so on. If any of those parameters are a failure state, then the entire
+creation process will fail, and the function will return the first failure
+state. If all of the parameters are successful, the value will be unwrapped and
+passed to the `init` function.
+
+### Safely pulling values from JSON
 
-In order to help with the decoding process, Argo introduces two new operators
-for parsing a value out of the JSON:
+In the example above, we showed some non-existent parameters (`paramOne`, etc), but
+one of Argo's main features is the ability to help you grab the real parameters
+from the JSON structure in a way that is type-safe and concise. You don't need
+to manually check to make sure that a value is non-nil, or that it's of the
+right type. Argo leverages Swift's expressive type system to do that leaving
+lifting for you. To help with the decoding process, Argo introduces two new
+operators for parsing a value out of the JSON:
 
 - `<|` will attempt to parse a single value from the JSON
 - `<||` will attempt to parse an array of values from the JSON
 
-The usage of these operators is the same regardless:
+The usage of these operators is simple:
+
+- `json <| "name"` is analogous to `json["name"]`
+- `json <|| "posts"` is analogous to `json["posts"]`
```

Comment:
Should this make a note that `posts` in this case is expected to be a JSON array?